app/controllers/recipients_controller.rb

```diff
@@ -0,0 +1,41 @@
+class RecipientsController < ApplicationController
```

Comment:
ok, I split these apart mainly due to routing. I took a look at how tmwsd's navigation vs how the messages were actually read. It used different routes for message admin vs message consumption. I went with a similar approach, which lead to some ugliness with endpoint config.

{code}
  get 'recipients/:id' => 'recipients#show'
  get 'recipients/:id/authenticate' => 'recipients#authenticate'
  post 'recipients/:id' => 'recipients#authenticatedShow'
{code} 

I liked this separation because the recipient view and authentication flow was distinct from that of the admin.